src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -288,6 +287,19 @@ public static boolean isItemSimilar(@Nullable ItemStack item, @Nullable ItemStac
 
                 ItemMetaSnapshot meta = ((SlimefunItemStack) sfitem).getItemMetaSnapshot();
                 return equalsItemMeta(itemMeta, meta, checkLore);
+            } else if (sfitem instanceof ItemStackWrapper && sfitem.hasItemMeta()) {
+                // Slimefun items may be ItemStackWrapper's in the context of cargo so let's try to do
+                // an ID comparison before meta comparison
+                ItemMeta possibleSfItemMeta = sfitem.getItemMeta();
+                Optional<String> possibleSfItemId = Slimefun.getItemDataService().getItemData(possibleSfItemMeta);
+                Optional<String> itemId = Slimefun.getItemDataService().getItemData(itemMeta);
+
+                if (possibleSfItemId.isPresent() && itemId.isPresent()) {
+                    return possibleSfItemId.get().equals(itemId.get());
+                } else {
+                    return equalsItemMeta(itemMeta, possibleSfItemMeta, checkLore);
+                }
```

Comment:
Now that I have time to look at this a little closer, I spot a way we can optimize this further.
If `possibleSfItemMeta` has no id, we don't need to extract the id from `itemMeta`, reducing it by one further `ItemStack#getItemMeta()` call.

Not the prettiest but you get the idea

```suggestion
                Optional<String> possibleSfItemId = Slimefun.getItemDataService().getItemData(possibleSfItemMeta);

                if (possibleSfItemId.isPresent()) {
                    Optional<String> itemId = Slimefun.getItemDataService().getItemData(itemMeta);

                    if (itemId.isPresent()) {
                        return possibleSfItemId.get().equals(itemId.get());
                    }
                }

                return equalsItemMeta(itemMeta, possibleSfItemMeta, checkLore);
```

Reply:
Scratch the above.
To declutter this:

We should implement a `CustomItemDataService#compare(...)` method.

Reply:
I'll make a follow-up commit.